### PR TITLE
[#7534] Separate DApp & chat tabs in mobile navigation

### LINKED ISSUE #7534: Separate DApp & chat tabs in mobile navigation
### User Story

As a user, I want to discover the DApps/browser in Status more easily and have separation between DApps and chat. 

### Description

*Type*: Feature

*Summary*: From a [previous issue](https://github.com/status-im/status-react/issues/7030), this is a long requested change by users. User problems to address are (1) difficulty locating DApps/browser in current nav, and (2) [distaste for clutter on home screen](https://github.com/status-im/status-react/issues/6675) between browser items & chat. 

```
The proposed solution addresses ongoing user feedback and findings from UXR such as:

* Path to switching modes (e.g. Wallet -> DApps) is unclear
* Lack of consistent UI element to refer to for app navigation
* Unclear how to start a new chat with a new or existing contact
* Separating DApps entry point from a submenu and bringing back to the main nav bar
```

#### Expected behavior
DApps & chat are their own tabs in the [new bottom nav bar](https://github.com/status-im/status-react/pull/7483), per these [Figma designs](https://www.figma.com/file/deXIBZm6fU1CCL0voYbfTujz/Navigation?node-id=1%3A2).

DApps tab takes users to existing `Open DApp` screen.

#### Actual behavior
Home screen mixes DApps & chat items.

### PR DESCRIPTION
**Pull request by flexsurfer**
FIRST ITERATION

image

fixes #7534

### CODE REVIEW

File: src/status_im/ui/screens/browser/open_dapp/views.cljs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+(defn- render-dapp [{:keys [dapp-url recent?] :as dapp}]
+  (if recent?
+    [list-item dapp]
+    [contact-view/contact-view {:contact             dapp
+                                :on-press            #(re-frame/dispatch [:browser.ui/open-dapp-button-pressed dapp-url])
+                                :show-forward?       true
+                                :accessibility-label :dapp-item}]))
+
+(defn list-header [empty?]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/browser/open_dapp/views.cljs:40]: not used in this PR , styles will be moved in next PR

File: src/status_im/ui/components/bottom_bar/core.cljs [hunk 2]
```diff
@@ -33,11 +33,11 @@
...
     :accessibility-label :home-tab-button}
-   #_{:nav-stack           :dapp-stack
-      :content             {:title (i18n/label :t/dapp)
-                            :icon  :main-icons/dapp}
-      ;;:count-subscription  :chats/unread-messages-number
-      :accessibility-label :dapp-tab-button}
+   {:nav-stack           :browser-stack
+    :content             {:title (i18n/label :t/dapp)
+                          :icon  :main-icons/dapp}
+    ;;:count-subscription  :chats/unread-messages-number
```
Comment (jeluard) [inline src/status_im/ui/components/bottom_bar/core.cljs]: Remove?

File: src/status_im/ui/components/list_item/styles.cljs [hunk 3]
```diff
@@ -13,7 +13,7 @@
             {:line-height 22 :font-size 15}
             {:line-height 20 :font-size 17})
     subtitle
-    (assoc :font-weight 500 :font-size 15)))
+    (assoc :font-weight "500" :font-size 15)))
```
Comment (jeluard) [inline src/status_im/ui/components/list_item/styles.cljs:16]: Any reason why?

File: src/status_im/ui/components/list_item/views.cljs [hunk 4]
```diff
@@ -18,7 +18,9 @@
       ;;Image
       (when image
         [react/view {:margin-left 16}
-         [image]])
+         (if (vector? image)
```
Comment (jeluard) [inline src/status_im/ui/components/list_item/views.cljs:21]: What's the use case for this?

File: src/status_im/ui/screens/home/views/inner_item.cljs [hunk 5]
```diff
@@ -19,7 +19,11 @@
             [status-im.ui.components.icons.vector-icons :as vector-icons]
             [status-im.ui.components.chat-icon.screen :as chat-icon.screen]
             [status-im.ui.components.common.common :as components.common]
-            [status-im.browser.core :as browser]))
+            [status-im.ui.components.list-item.views :as list-item]
+            [status-im.browser.core :as browser]
+            [clojure.string :as string]
+            [status-im.ui.components.chat-icon.screen :as chat-icon]
```
Comment (jeluard) [inline src/status_im/ui/screens/home/views/inner_item.cljs]: Not all used?

File: src/status_im/ui/screens/routing/browser_stack.cljs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+(ns status-im.ui.screens.routing.browser-stack)
+
+(def browser-stack
```
Comment (jeluard) [inline src/status_im/ui/screens/routing/browser_stack.cljs:3]: Maybe add a comment to detail what this is?

**Review by jeluard — APPROVED** [review]

File: src/status_im/ui/components/bottom_bar/core.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/status_im/ui/components/bottom_bar/core.cljs]: yes

File: src/status_im/ui/components/list_item/styles.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/status_im/ui/components/list_item/styles.cljs:16]: idk why but in current version of app it crashes in run time

File: src/status_im/ui/components/list_item/views.cljs (hunk 4 above)
Comment (flexsurfer) [reply src/status_im/ui/components/list_item/views.cljs:21]: image can be provided as  a render function or as a vector , for example `:image render` or `:image [render]`

File: src/status_im/ui/components/list_item/styles.cljs (hunk 3 above)
Comment (errorists) [reply src/status_im/ui/components/list_item/styles.cljs:16]: it takes an enum from `('normal', 'bold', '100', '200', '300', '400', '500', '600', '700', '800', '900')`, passing a number won't work

**Review by errorists — APPROVED** [review]
It's not pretty, but I guess gets the job done until we have the dapp store running. One comment, I recall the tab bar item should be labelled plural 'ÐApps'

**Comment by asemiankevich** [comment]
@antdanchenko we need to adjust e2e here a bit , to access dapps not  from plus button rather than from a tab. Or maybe @Serhy or @churik want to assist here?

**Comment by asemiankevich** [comment]
@flexsurfer did we remove dapps info (with dapp description) intentionally?

**Comment by flexsurfer** [comment]
@asemiankevich yes we did

**Comment by flexsurfer** [comment]
@asemiankevich @antdanchenko please consider that this is a temporary step, and in a next PR list of dapps will be removed 
image

**Comment by flexsurfer** [comment]
https://www.figma.com/file/MYWmd1buvc2AMvUmFP9w42t5/Discovery?node-id=35%3A420

**Comment by jeluard** [comment]
@flexsurfer Do we plan to release this intermediary step?

**Comment by errorists** [comment]
[here's the entire roadmap as told in mockups :) ](https://www.figma.com/file/MYWmd1buvc2AMvUmFP9w42t5/Discovery?node-id=932%3A1)

the dapp profile, we agreed to remove it since it had very little utility and was a step in between navigating to the dapp itself from the list

**Comment by flexsurfer** [comment]
@jeluard i believe we are fine to merge / release this version, because store dapp development will take some time for sure

**Comment by Serhy** [comment]
Updated critical autotests and confirm they pass now ( https://github.com/status-im/status-react/pull/7719#issuecomment-472353060 )
Tested builds on iPhoneX (iOS 12.1.4) and Android 8.1 (Samsung Galaxy J7)

**Comment by rachelhamlin** [comment]
@flexsurfer you mean fine to release the version in your original post? 

image

**Comment by flexsurfer** [comment]
@rachelhamlin yes, do you think it's not fine?

**Comment by rachelhamlin** [comment]
Porting conversation from Status here :) 

To me it feels least disorienting and most user friendly to release *only* the nav change first. Leave DApp descriptions as they are, don't add the recent category. And then introduce the DApp store in all its glory only once the MVP is reviewed and ready.

**Comment by oskarth** [comment]
What's the difference between this and the Discovery tab we had before? It'd be useful to understand why we removed it, and what we have learned that led us to add it back again. Right now my first impression is that we are going back and forth, and it isn't clear to me why or what has changed.

**Comment by rachelhamlin** [comment]
What do you mean about the Discovery tab @oskarth? 

@flexsurfer makes a fair point about losing browser history if we have no `Recent` category. I can see the case for that.

**Comment by errorists** [comment]
Alternatively, we can add a plus button over here as well, like we have in chat, and once tapped open up the existing 'Open Dapp' view with categories and address bar like nothing changed. then this entire view is recents, similar to chat tab in appearance and function.

**Comment by oskarth** [comment]
> What do you mean about the Discovery tab @oskarth?

A year ago we had a fourth tab that said "Discovery", and it seems to be doing what this is doing _and more_. Then we removed it, and now we are adding it back again. I realize people responsible for this are different, but here are some questions I'd like to see the answers to:

1. Why did we remove Discovery tab in the first place? What was the rationale?
2. Has that rationale changed? In what way?
3. What is different about this tab and the old Discovery design and tab?
4. What have we learned in the last year that led us to these changes?
5. If we re-introduce Discovery in the future, how will that change the answers to above questions? (I realize it was building on a shakey foundation, but the concern still stands - how do we ensure we are going *forward* and not back and forth?).

PS I think it's great that we separate these out, it's just worrisome that it seems like we are going back and forth without understanding why we are changing things (at least from my naive POV, I probably miss a lot of context, hence the questions) DS

**Comment by hesterbruikman** [comment]
> * Why did we remove Discovery tab in the first place? What was the rationale?

At the time I understood from Chad main consideration to remove Discover was related to getting an MVP out for Beta. There were too many discussions and open ends around the tab that it was concluded to be better to leave out.

> * Has that rationale changed? In what way?

We have some well thought out designs now for the behavior of a seperate tab. We've also seen repeatedly (this as come up in every test since January last year, seen a need for a more clear distinction between Chat and DApps. Will copy below a summary I sent to Jonny earlier this week.

> * What is different about this tab and the old Discovery design and tab?

I don't know the old Discover tab well enough:/ Maybe @Anna can better compare.

> * What have we learned in the last year that led us to these changes?

Think the summary I sent to Jonny answers this
```
"We've made the navigation persistent and will give DApps their own tab in the coming weeks. With these changes you will always know where you are, be able to easily multitask between tabs and find DApps more easily.

These changes are a response to learnings from UX testing as well as direct feedback.

* We've repeatedly observed it being difficult for people to find DApps, regardless of whether they were familiar with the concept or DApps label and even if they had used the plus-button and had had 'some exposure' to DApps.

Next to this, we've seen that it is difficult to navigate between
- DApp and Wallet (for example to quickly check ones balance)
- Wallet, Profile and Home (where chats and DApps lived)

We've also repeatedly received explicit feedback on usability of the navigation. e.g. _What could have made your experience better?_
- "Better UX when navigating around pages within the browser, and tabs." 
- "having a tab only for dapps"
```

**Comment by rachelhamlin** [comment]
@hesterbruikman covered it all, I think. :) 

Hard to speak to the decision on Discover tab, but from what I understand, Discover was a beast of a feature. This is really just about navigation and is rooted in solid research.

**Comment by goranjovic** [comment]
@hesterbruikman @rachelhamlin 

> Discover was a beast of a feature. 

This sums it up nicely. 

The  intention was to propagate discoverable items throughout the network by pushing them to your contacts, then they'd automatically push it to theirs and so on. These items were initially meant to be not just dapps but also public chats and "user statuses" (something between your profile headline and a decentralized tweet). That's massive research effort alone, especially with regard to privacy as content propagation would be a metadata leak if not done right. 

That said, this PR merely moves the (hardcoded?) dapp list that we already have from a nested menu to its own first level tab. Does that sound right? @flexsurfer

**Review by goranjovic — APPROVED** [review]

**Comment by flexsurfer** [comment]
> That said, this PR merely moves the (hardcoded?) dapp list that we already have from a nested menu to its own first level tab. Does that sound right? @flexsurfer

yes, that's exactly what this PR is about!

next step (next PR) is to move this hardcoded! list to dapp outside status

**Comment by flexsurfer** [comment]
image

**Review by rachelhamlin — APPROVED** [review]

**Pull request merged**